src/filters/global.filter.ts

```diff
@@ -1,47 +1,47 @@
-import { ArgumentsHost, Catch, ExceptionFilter, Logger } from '@nestjs/common';
+import { ArgumentsHost, Catch, ExceptionFilter, HttpException, Logger } from '@nestjs/common';
 import { Request, Response } from 'express';
-import { GlobalException } from 'src/exceptions/global.exception';
-import ERRORS from 'src/errors/errors.config';
+import { I18nService } from 'nestjs-i18n';
 import ErrorsTypes from 'src/errors/errors.enum';
+import { GlobalException } from 'src/exceptions/global.exception';
+import { translateErrorMessage } from 'src/helpers/translateErrorMessage.helper';
 
 @Catch()
 export class GlobalExceptionFilter implements ExceptionFilter {
-  constructor(private readonly logger: Logger) {}
+  constructor(
+    private readonly logger: Logger,
+    private readonly i18n: I18nService
+  ) {}
 
   catch(exception: GlobalException | any, host: ArgumentsHost) {
     const ctx = host.switchToHttp();
     const response = ctx.getResponse<Response>();
+    const defaultErrorMessage = translateErrorMessage(this.i18n, ErrorsTypes.DEFAULT);
+    const defaultResponse = {
+      success: false,
+      error: { statusCode: 500, message: defaultErrorMessage },
+    };
+
     try {
+      const knownException = exception instanceof HttpException;
       const request = ctx.getRequest<Request>();
-      const status = exception.getStatus();
 
-      const errorObj = exception.getResponse();
+      const status = knownException ? exception.getStatus() : 500;
+      const errorObj = knownException ? exception.getResponse() : exception;
+      const key = knownException ? errorObj?.key || errorObj?.message : ErrorsTypes.DEFAULT;
+      const message = knownException ? translateErrorMessage(this.i18n, key) : exception.message;
```

Comment:
looks like you repeat yourself
make default assignment first
let logString =  `method: ${request.method}, to: ${request.originalUrl}, status: 500, errorMessage: ${exception.getResponse().key}, cause: ${errorObj.message}`

if (exception instanceof HttpException) {
   logString = ...
}


of make particular assignment 

      let status =  500 (should be enum or const here)
      let errorObj = exception;
      let key = ErrorsTypes.DEFAULT;
      let message =  exception.message;

if ( exception instanceof HttpException) {
status = ....

Reply:
i changed, is it ok now?